avm/ptn/app/iaas-vm-cosmosdb-tier4/main.bicep

```
@@ -0,0 +1,828 @@
+metadata name = 'IaaS VM with CosmosDB Tier 4'
+metadata description = 'Creates an IaaS VM with CosmosDB Tier 4 resiliency configuration.'
+
+@description('Required. Name of the solution which is used to generate unique resource names.')
+param name string
+
+@description('General. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('General. Tags for all resources. Should include standard tags like Environment, Owner, CostCenter, etc.')
+@metadata({
+  example: {
+    Environment: 'Production'
+    Owner: 'TeamName'
+    CostCenter: 'IT'
+    Application: 'MyApp'
+  }
+})
+param tags object?
```

Should ideally be typed. As it doesn't really matter where to take the tags type from as they're all the same, you could for example use
```suggestion
param tags resourceInput<'Microsoft.Network/networkSecurityGroups@2024-07-01'>.tags?
```